homeassistant/components/heos/__init__.py

```diff
@@ -0,0 +1,41 @@
+"""Denon HEOS Media Player."""
+
+import voluptuous as vol
+
+from homeassistant.components.media_player.const import (
+    DOMAIN as MEDIA_PLAYER_DOMAIN)
+from homeassistant.const import CONF_HOST, EVENT_HOMEASSISTANT_STOP
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+DOMAIN = 'heos'
+REQUIREMENTS = ['aioheos==0.4.0']
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass: HomeAssistantType, config: ConfigType):
+    """Set up the HEOS component."""
+    from aioheos import AioHeosController
+
+    host = config[DOMAIN][CONF_HOST]
+    controller = AioHeosController(hass.loop, host)
+    await controller.connect()
```

Comment:
What happens if we can't connect? Is there a time out? Otherwise we'll hold up home assistant from finishing set up phase.